editor/src/components/canvas/controls/select-mode/select-mode.spec.browser2.tsx

```diff
@@ -640,6 +640,218 @@ describe('Select Mode Double Clicking', () => {
   })
 })
 
+describe('Select Mode Double Clicking With Fragments', () => {
+  // The below tests are similar to tests in 'Select Mode Double Clicking', but
+  // they use a test project which contain components which renders fragment root
+  // elements. These are special cases because these fragments and their components
+  // do not appear in the dom.
+
+  before(() => {
+    viewport.set(2200, 1000)
+  })
+
+  it('One double clicks to select Card Instance', async () => {
+    // prettier-ignore
+    const desiredPath = EP.fromString(
+      'sb' +            // Skipped as it's the storyboard
+      '/scene-2' +      // Skipped because we skip over Scenes
+      '/Card-instance', // <- First double click
+    )
+
+    const renderResult = await renderTestEditorWithCode(
+      TestProjectAlpineClimb,
+      'await-first-dom-report',
+    )
+
+    const cardSceneRoot = renderResult.renderedDOM.getByTestId('card-scene')
+    const cardSceneRootBounds = cardSceneRoot.getBoundingClientRect()
+
+    const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    const fireDoubleClickEvents = createDoubleClicker()
+
+    const doubleClick = async () => {
+      await act(async () => {
+        fireDoubleClickEvents(
+          canvasControlsLayer,
+          cardSceneRootBounds.left + 130,
+          cardSceneRootBounds.top + 220,
+        )
+      })
+    }
+
+    await doubleClick()
+
+    expect(renderResult.getEditorState().editor.selectedViews).toEqual([desiredPath])
+  })
+
+  it('Two double clicks to select Card Scene Root', async () => {
+    // prettier-ignore
+    const desiredPath = EP.fromString(
+      'sb' +             // Skipped as it's the storyboard
+      '/scene-2' +       // Skipped because we skip over Scenes
+      '/Card-instance' + // <- First double click
+      ':Card-Root',      // <- Second double click, as the instance is automatically focused by the scene
```

Comment:
oh this is _very nice_ and descriptive

Reply:
Oh thanks :)

Reply:
I agree! I always copy paste from the best ones!

Reply:
> Oh thanks :)

somehow I completely missed #2436, but kudos Rheese, it's _very nice_

Reply:
this is very easy to read!